### PR TITLE
Failed to start if expected frontends not matched

### PR DESCRIPTION
**Pull request by chbatey**
Doh, all of the other demo comments were actually in umc-platform so should have just snook this in the last one

### CODE REVIEW

File: ingress/elb/elb.go [hunk 1]
```diff
@@ -119,6 +119,11 @@ func (e *elb) Attach() error {
 	prometheus.Register(attachedFrontendGauge)
 	attachedFrontendGauge.Set(float64(registered))
 	e.registeredFrontends = registered
+
+	if registered != e.expectedFrontends {
+		log.Warn("Expected ELBs: %d actual: %d", e.expectedFrontends, registered)
```
Comment (balooo) [inline ingress/elb/elb.go]: should be `log.Warnf`

**Comment by balooo** [comment]
Also, if we're checking that registered == expected on startup,  I don't understand why we are checking it as part of the Health check.   If it fails on startup, it won't startup.  If we want to check those values on a healthcheck poll, shouldn't we re query the number of registered to make sure it still matches?

**Comment by jsravn** [comment]
This made me think we should just fail if it can't attach. Because the deployment will fail if it never goes healthy (it waits for the pod to be ready, otherwise deployment times out). So we may as well fail the startup if it doesn't match rather than just leaving it to the health check.

**Comment by jsravn** [comment]
Similarly for the Health check, we should probably periodically check the frontends are still attached. But that can be a separate task. (And maybe it's too edge case to worry about..?)

**Comment by chbatey** [comment]
yeah let's just return an errro

**Comment by chbatey** [comment]
There's still value in logging this even if kubelet does just kill it as we will get to see why it  killed it from the last container's logs

**Comment by jsravn** [comment]
If it returns an error on `Attach` it will get logged by `main.go`.

**Comment by chbatey** [comment]
I have updated this to return an error from Start() if the number is not matched. Ready for review.

**Comment by jsravn** [comment]
lgtm

**Pull request merged**